openreview/venue/process/paper_recruitment_pre_process.js

```diff
@@ -0,0 +1,69 @@
+async function process(client, edit, invitation) {
+
+  const committeeInvitedId = invitation.content.committee_invited_id.value
+  const committeeName = invitation.content.committee_name.value
+  const inviteAssignmentInvitation = invitation.content.invite_assignment_invitation_id.value
+  const hash_seed = invitation.content.hash_seed.value
+  
+  const { groups: domainGroups } = await client.getGroups({ id: invitation.domain })
+  const domain = domainGroups[0]
+  const venueId = domain.id
+  const submissionVenueId = domain.content.submission_venue_id.value
+  
+  const note = edit.note
+  const user = decodeURIComponent(note.content.user.value)
+  const hashkey = crypto.createHmac('sha256', hash_seed)
+                      .update(user)
+                      .digest('hex');
+
+  if (hashkey != note.content.key.value) {
+    return Promise.reject(new OpenReviewError({ name: 'Error', message: 'Wrong key, please refer back to the recruitment email' }))
+  }
+
+  const { notes } = await client.getNotes({ id: note.content.submission_id.value })
+  const submission = notes[0]
+
+  const { count: edgeCount, edges } = await client.getEdges({ invitation: inviteAssignmentInvitation, head: submission.id, tail: note.content.user.value })
+  let inviteEdges = edges
+
+  if (edgeCount == 0) {
+    const { profiles} = await client.getProfiles(user.startsWith('~') ? { id: user } : { email: user })
+    if (profiles.length > 0) {
+      const { count: edgeCount, edges } = await client.getEdges({ invitation: inviteAssignmentInvitation, head: submission.id, tail: profiles[0].id })
+      inviteEdges = edges
+      if (edgeCount == 0) {
+        return Promise.reject(new OpenReviewError({ name: 'Error', message: 'Invitation no longer exists. No action is required from your end.' }))
+      }
+    }
+  }
+
+  const { count: groupCount } = await client.getGroups({ id: committeeInvitedId, member: user })
```

Comment:
If you want to check that `user` is a member of the `committeeInvitedId` group, you can use `members` instead which does not calculate transitive members and is therefore more efficient.

```suggestion
  const { count: groupCount } = await client.getGroups({ id: committeeInvitedId, members: user })
```

Reply:
I do want to check the transitive members